File: 66.Plus One.py
class Solution(object):
    def plusOne(self, digits):
        """
        :type digits: List[int]
        :rtype: List[int]
        """
        res = []
        plus = 1
        for i in range(len(digits)-1, -1, -1):
            digit = digits[i] + plus
            if digit>9:
                plus = 1
            else:
                plus = 0
            res.append(digit%10)
        if plus == 1 :
            res.append(1)
        return res[::-1]
        

File: 66.Plus One.cpp
class Solution {
public:
    vector<int> plusOne(vector<int>& digits) {
        vector<int> res;
        int plus = 1;
        int digit = 0;
        for(int i = digits.size()-1; i>=0; i--){
            digit = digits[i] + plus;
            if(digit==10){
                plus = 1;
                res.push_back(0);
            }
            else{
                plus = 0;
                res.push_back(digit); 
            }
        }
        if(plus==1) res.push_back(1);
        reverse(res.begin(), res.end());
        return res;
    }
};
